frontend/app/services/pdf-service.ts

```diff
@@ -0,0 +1,46 @@
+import { z } from 'zod';
+
+import { getEnv } from '~/utils/env.server';
+import { getLogger } from '~/utils/logging.server';
+
+const pdfSchema = z.object({
+  fileStream: z.string().optional(),
+  referenceId: z.string().optional(),
+  id: z.string().uuid().optional(),
+});
+
+export type PdfInfo = z.infer<typeof pdfSchema>;
+
+function createPdfService() {
+  const logger = getLogger('pdf-service.server');
+  const { CCT_API_BASE_URI } = getEnv();
+
+  /**
+   *
+   * @param { referenceId }
+   * @returns returns the pdf based off  @param referenceId
+   */
+  async function getPdf(referenceId: string) {
+    const url = `${CCT_API_BASE_URI}/cct/letters/${referenceId}`;
+    const response = await fetch(url);
+
+    if (response.ok) return pdfSchema.parse(await response.json());
+    if (response.status === 404) return null;
+
+    logger.error('%j', {
+      message: 'Failed to fetch data',
+      status: response.status,
+      statusText: response.statusText,
+      url: url,
+      handlersresponseBody: await response.text(),
+    });
+
+    throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
```

Comment:
As with the other service, I would recommend handling the error state early, so you leave room in the function to do any post-processing of data (should you need to).

Reply:
So in this case, you would do something like:

``` typescript
if (!response.ok) {
  if (response.status === 404) { return null; }
  // throw...
}

// rest of function
```

Reply:
This could be applied as a service refactoring task.

Reply:
Are you suggesting in a future PR?

Reply:
> Are you suggesting in a future PR?

Yes